### PR TITLE
convert interfaces

### PR DESCRIPTION
**Pull request by yibeichan**
hello @effigies, I converted `fitlins/fitlins/interfaces/abstract.py` into `pydra/tasks/fitlins/interfaces/abstract.py`  since the original `nistats.py` cited this `abstract.py` a lot. 
Would you like to take a look at it? 
1. I'm not sure whether I specified `type` for each field correctly. I added annotations there 
2. may I ask why the first level estimator output is called `EstimatorOutputSpec` not `FirstLevelEstimatorOutputSpec` while we do have a `SecondLevelEstimatorOutputSpec`

Thanks!

### CODE REVIEW

**Comment by yibeichan** [comment]
sorry, I have a question, why I have added all packages needed in `requirements.txt`, I still got errors `no module named xxx ` from the test

**Comment by effigies** [comment]
Apologies for the slow responses, I'm only just getting back to work after the conference.

> sorry, I have a question, why I have added all packages needed in `requirements.txt`, I still got errors `no module named xxx ` from the test

We declare dependencies here:

https://github.com/nipype/pydra-fitlins/blob/1cda476ddb915daafdfe848bd165fa4447f38e0d/setup.cfg#L25-L26

For dependencies only required for testing, we can add them here:

https://github.com/nipype/pydra-fitlins/blob/1cda476ddb915daafdfe848bd165fa4447f38e0d/setup.cfg#L44-L50

File: pydra/tasks/fitlins/utils/config.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+"""
+TODO: change nipype to pydra
```
Comment (effigies) [inline pydra/tasks/fitlins/utils/config.py]: Let's just drop this file for now.

File: pydra/tasks/fitlins/utils/bids.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        load_all_specs(all_specs, specs, child.destination, **child.filter)
+
+
+def collect_participants(layout, participant_label=None, strict=False):
+    """
+    List the participants under the BIDS root and checks that participants
+    designated with the participant_label argument exist in that folder.
+    Returns the list of participants to be finally processed.
+    Requesting all subjects in a BIDS directory root:
+    >>> collect_participants(layout)
```
Comment (yibeichan) [inline pydra/tasks/fitlins/utils/bids.py]: hello @effigies I keep getting errors here from the test, saying that `NameError: name 'layout' is not defined`. I don't fully understand why those lines as comments will be examined in the test. I guess if I remove those lines, the error should be resolved?

Comment (yibeichan) [reply pydra/tasks/fitlins/utils/bids.py]: oh I got it, it's doctest....

Comment (yibeichan) [reply pydra/tasks/fitlins/utils/bids.py]: I found a similar function here https://www.nipreps.org/niworkflows/1.6/_modules/niworkflows/utils/bids.html, this one seems to have a better documentation.
So `NameError: name 'layout' is not defined` can be fixed if we setup something similar to `data_dir_canary()` beforehand? 
Also, at the beginning of this file, https://github.com/poldracklab/fitlins/blob/dev/fitlins/utils/bids.py
we have 
"""
Utilities to handle BIDS inputs
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Fetch some test data
    >>> import os
    >>> from niworkflows import data
    >>> data_root = data.get_bids_examples(variant='BIDS-examples-1-enh-ds054')
    >>> os.chdir(data_root)
"""
However, `get_bids_examples` this function doesn't exist in the latest niworkflows. Do you want to specify the version of `niworkflows` to be installed, or we have an alternative function?

Comment (effigies) [reply pydra/tasks/fitlins/utils/bids.py]: I don't think fitlins depends on niworkflows, and this package definitely shouldn't.

Comment (effigies) [reply pydra/tasks/fitlins/utils/bids.py]: I think removing broken doctests for now is probably the best plan.

Comment (yibeichan) [reply pydra/tasks/fitlins/utils/bids.py]: Thank you and sorry I just saw this! I'll fix it soon. 
Also, I just get a better understanding of how to use `typing` represent `traits.Enum/Either`. I'll fix some of that too!

File: pydra/tasks/fitlins/utils/config.py (hunk 1 above)
Comment (yibeichan) [reply pydra/tasks/fitlins/utils/config.py]: removed!

**Comment by yibeichan** [comment]
okay... the test failed again. But this time I know why! So I have something like `list[dict]` for annotation; however, 'type' object is not subscriptable until python 3.9. That's probably the reason I failed test(3.7) and test(3.8)....
So, how should we deal with python version difference?

File: pydra/tasks/fitlins/utils/bids.py (hunk 2 above)
Comment (yibeichan) [reply pydra/tasks/fitlins/utils/bids.py]: fixed this one (and we have other new problems) so I'm gonna resolve this conversation

**Comment by yibeichan** [comment]
> okay... the test failed again. But this time I know why! So I have something like `list[dict]` for annotation; however, 'type' object is not subscriptable until python 3.9. That's probably the reason I failed test(3.7) and test(3.8).... So, how should we deal with python version difference?

@djarecka here it is

**Comment by djarecka** [comment]
I think we can remove 3.7 from GA, so it should work. Will merge and fix the testing in another PR

**Pull request merged**